crates/mpt/src/node.rs

```diff
@@ -375,13 +375,13 @@ impl TrieNode {
             Self::Leaf { prefix, value } => {
                 // Encode the leaf node's header and key-value pair.
                 Header { list: true, payload_length }.encode(out);
-                prefix.encode_path_leaf(true).as_slice().encode(out);
+                prefix.pack().as_slice().encode(out);
```

Comment:
This shouldn't work - [`Nibbles::pack`](https://docs.rs/nybbles/latest/nybbles/struct.Nibbles.html#method.pack) just packs the nibbles into a slice. `alloy_trie` has this function now, should be able to use it from there: [`alloy_trie::nodes::encode_path_leaf`](https://docs.rs/alloy-trie/latest/alloy_trie/nodes/fn.encode_path_leaf.html)